servers/dsub/jobs/test/base_test_cases.py

```diff
@@ -438,3 +468,174 @@ def test_query_jobs_submission_pagination(self):
                     page_token=response.next_page_token,
                     extensions=ExtendedQueryFields(submission=min_time)),
                 [job2])
+
+        def status_counts_to_dict(self, status_counts):
+            status_counts_dict = {}
+            for status_count in status_counts.counts:
+                status_counts_dict[status_count.status] = status_count.count
+            return status_counts_dict
+
+        def aggregation_response_to_dict(self, aggregation_response):
+            # Flatten the nested aggregation response to a dict that is easy to compare equality
+            aggregation_response_dict = {}
+            aggregation_response_dict['summary'] = self.status_counts_to_dict(
+                aggregation_response.summary)
+
+            for aggregation in aggregation_response.aggregations:
+                entry_dict = {}
+
+                for entry in aggregation.entries:
+                    entry_dict[entry.label] = self.status_counts_to_dict(
+                        entry.status_counts)
+
+                aggregation_response_dict[aggregation.key] = entry_dict
+
+            return aggregation_response_dict
+
+        def assert_aggregation_response_equal(self, aggregation_response,
+                                              userId, name):
+            aggregation_response_dict = self.aggregation_response_to_dict(
+                aggregation_response)
+            status_counts_total = {
+                'Running': 3,
+                'Failed': 1,
+                'Succeeded': 1,
+                'Aborted': 1
+            }
+
+            status_counts_group = {
+                'Running': 1,
+                'Failed': 1,
+                'Succeeded': 1,
+                'Aborted': 1
+            }
+
+            status_counts_single = {'Running': 1}
+
+            expected_aggregations_response_dict = {
```

Comment:
`assert_aggregation_response_equal` should be a generic function. So it shouldn't have a specific hard-coded response here that's tied to your test. Would prefer this were defined in the test function, then passed as a parameter to `assert_aggregation_response_equal`.

Would also be nice if that parameter were of the same form (AggregationResponse, rather than dict), but fine to keep it a dict if using the model is to verbose. If you want to leave the expected response as a dict, I'd just remove this function, convert the test response to a dict in your test method and call assert_equals() directly there.